### PR TITLE
Add IXamlParameterInfo

### PR DESCRIPTION
**Pull request by maxkatz6**
See https://github.com/AvaloniaUI/Avalonia/pull/14612

### CODE REVIEW

File: src/XamlX.IL.Cecil/CecilMethod.cs [hunk 1]
```diff
@@ -75,7 +92,7 @@ MethodReference MakeRef(bool transform)
 
             public IReadOnlyList<IXamlType> Parameters =>
                 _parameters ?? (_parameters =
-                    Reference.Parameters.Select(p => TypeSystem.Resolve(p.ParameterType)).ToList());
+                    Reference.Parameters.Select(p => new CecilParameterInfo(TypeSystem, p).ParameterType).ToList());
```
Comment (kekekeks) [inline src/XamlX.IL.Cecil/CecilMethod.cs]: Won't it be better to pre-cache  CecilParameterInfo[] array and then use it as dependency here?

**Review by kekekeks — APPROVED** [review]

File: src/XamlX.IL.Cecil/CecilMethod.cs (hunk 1 above)
Comment (maxkatz6) [reply src/XamlX.IL.Cecil/CecilMethod.cs]: Maybe? Probably a low profit change

**Comment by maxkatz6** [comment]
@kekekeks this PR can be merged now. Another PR might be reviewer later, but I will independently push XamlX.Roslyn fixes to make master compatible with this IXamlParameterInfo, until another PR is merged.

File: src/XamlX.IL.Cecil/CecilMethod.cs (hunk 1 above)
Comment (maxkatz6) [reply src/XamlX.IL.Cecil/CecilMethod.cs]: Pre-cached these parameters.

**Comment by maxkatz6** [comment]
@kekekeks updated this PR from the current master changes.

**Review by kekekeks — APPROVED** [review]

**Pull request merged**